src/plugins/techStackGeneric.nim

```diff
@@ -178,29 +179,22 @@ proc hostHasTechStack(scope: hostScope, proc_names: HashSet[string]): bool =
 proc scanDirectory(directory: string, category: string, subcategory: string) =
   if inFileScope[category][subcategory]:
     return
-  for filePath in walkDir(directory):
+  for kind, path in walkDir(directory):
     if inFileScope[category][subcategory]:
       break
-    if filePath.kind == pcFile:
-      scanFile(filePath.path, category, subcategory)
-      continue
-    if filePath.kind == pcDir:
-      scanDirectory(filePath.path, category, subcategory)
-      continue
+    if kind == pcFile:
```

Comment:
Not sure what the nim conventions are but maybe use switch here as it switched on the kind and there are no other conditions in the branches?